### PR TITLE
DOC-4340 more detail about connection pools and multiplexing

### PR DESCRIPTION
**Pull request by andy-stark-redis**
[DOC-4340](https://redislabs.atlassian.net/browse/DOC-4340)

[DOC-4340]: https://redislabs.atlassian.net/browse/DOC-4340?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/develop/connect/clients/java/jedis.md [hunk 1]
```diff
@@ -56,46 +56,36 @@ To include `Jedis` as a dependency in your application, edit the dependency file
...
-        }
+        // Code that interacts with Redis...
+
+        jedis.close();
     }
 }
 ```
 
-Because adding a `try-with-resources` block for each command can be cumbersome, consider using `JedisPooled` as an easier way to pool connections.
+When you have connected, you can check the connection by storing and
```
Comment (dwdougherty) [inline content/develop/connect/clients/java/jedis.md:76]:
```suggestion
After you have connected, you can check the connection by storing and
```

File: content/develop/connect/clients/pools-and-muxing.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+- kubernetes
+- clients
+description: Manage Redis connections efficiently
+linkTitle: Pooling/multiplexing
+title: Connection pools and multiplexing
+weight: 10
+---
+
+Redis example code generally opens a connection, demonstrates
+a command or feature and then closes. Real-world code typically
```
Comment (dwdougherty) [inline content/develop/connect/clients/pools-and-muxing.md:19]:
```suggestion
a command or feature, and then closes. Real-world code typically
```

File: content/develop/connect/clients/pools-and-muxing.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+## Connection pooling
+
+When you initialize a connection pool, the client opens a small number
+of connections and adds them to the pool.
+
+{{< image filename="/images/dev/connect/pool-and-mux/ConnPoolInit.drawio.svg" >}}
+
+Each time you "open" a connection
+from the pool, the client actually justs returns one of these existing
```
Comment (dwdougherty) [inline content/develop/connect/clients/pools-and-muxing.md:46]:
```suggestion
from the pool, the client returns one of these existing
```

File: content/develop/connect/clients/pools-and-muxing.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+{{< image filename="/images/dev/connect/pool-and-mux/ConnPoolInUse.drawio.svg" >}}
+
+When you later "close"
+the connection, the client puts it back into the pool of available
+connections without actually closing it.
+
+{{< image filename="/images/dev/connect/pool-and-mux/ConnPoolDiscon.drawio.svg" >}}
+
+If all connections in the pool are in use but the app needs more then
```
Comment (dwdougherty) [inline content/develop/connect/clients/pools-and-muxing.md:57]:
```suggestion
If all connections in the pool are in use but the app needs more, then
```

File: content/develop/connect/clients/pools-and-muxing.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+the client can simply open new ones as necessary. In this way, the client
```
Comment (dwdougherty) [inline content/develop/connect/clients/pools-and-muxing.md:58]:
```suggestion
the client can simply open new connections as necessary. In this way, the client
```

File: content/develop/connect/clients/pools-and-muxing.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+eventually finds the right number of connections to satisfy your
+app's demands.
+
+## Multiplexing
+
+Instead of pooling several connections, a multiplexer keeps a
+single connection open and uses it for all traffic between the
+client and the server. The "connections" returned to your code are
+simply to identify where to send the response data from your commands.
```
Comment (dwdougherty) [inline content/develop/connect/clients/pools-and-muxing.md:67]:
```suggestion
used to identify where to send the response data from your commands.
```

File: content/develop/connect/clients/pools-and-muxing.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+{{< image filename="/images/dev/connect/pool-and-mux/ConnMux.drawio.svg" >}}
+
+Note that it is not a problem if the multiplexer receives several commands close
+together in time. When this happens, the multiplexer can often combine the commands into a
+[pipeline]({{< relref "/develop/use/pipelining" >}}), which
+improves efficiency even more.
```
Comment (dwdougherty) [inline content/develop/connect/clients/pools-and-muxing.md:74]:
```suggestion
improves efficiency.
```

File: content/develop/connect/clients/python/redis-py.md [hunk 8]
```diff
@@ -232,6 +232,40 @@ The client will also flush the cache automatically
 if any connection (including one from a connection pool)
 is disconnected.
 
+## Connect with a connection pool
+
+For production usage, you should use a connection pool to manage
+connections rather than opening and closing connections individually.
+A connection pool maintains several open connections and reuses them
+efficiently. When you open a connection from a pool, the pool allocates you
```
Comment (dwdougherty) [inline content/develop/connect/clients/python/redis-py.md:240]:
```suggestion
efficiently. When you open a connection from a pool, the pool allocates
```

**Review by dwdougherty — APPROVED** [review]
LGTM.

**Comment by andy-stark-redis** [comment]
Thanks for catching those issues, @dwdougherty - much appreciated :-)

**Pull request merged**